### PR TITLE
Add ctest definitions for cloudsc variants

### PR DESCRIPTION
**Pull request by reuterbal**
Compiler vendor devtechs like to do a simple `ctest` after building the dwarf to check stuff works as expected. This adds a few sanity tests to the CMake to facilitate that.

Used the opportunity to also apply the CMake cleanup to the classic Fortran variants, which have somehow evaded that change...

### CODE REVIEW

File: src/cloudsc_loki/CMakeLists.txt [hunk 1]
```diff
@@ -266,6 +358,13 @@ if( HAVE_CLOUDSC_LOKI )
         DEFINITIONS ${CLOUDSC_DEFINITIONS}
     )
 
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-loki-scc-serial
+        COMMAND bin/dwarf-cloudsc-loki-scc
+        ARGS 1 1280 128
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 1
```
Comment (mlange05) [inline src/cloudsc_loki/CMakeLists.txt:366]: Same as with the standalone GPU version; "NVCOMPILER_ACC_CUDA_HEAPSIZE=64M" fixes it.

**Review by mlange05 — CHANGES_REQUESTED** [review]
Ok, this is looking really great, many thanks! The few issues I had with this are not showstoppers, so I'm marking this as "changes requested", but these should be trivial, and/or could be disabled/bypassed if needed.

The key ones are to force the CUDA_HEAPSIZE flag for both SCC variants, and somehow disabling the CLAW GPU runs (which seem to be broken in develop already).

I've also had trouble running the MPI ones behind `srun`, and I'm not sure if there is a good solution to pass additional arguments via flags to customise. If I've missed something obvious, a README entry might be nice to point those out. However, please ignore if this is cumbersome.

File: src/cloudsc_gpu/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +67,14 @@ if( HAVE_CLOUDSC_GPU_SCC )
             cloudsc-common-lib
         DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_SCC
     )
+
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-gpu-scc-serial
+        COMMAND bin/dwarf-cloudsc-gpu-scc
+        ARGS 1 1000 128
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
+        OMP 1
```
[the remark this answers is not in the document]
Comment (reuterbal) [reply src/cloudsc_gpu/CMakeLists.txt:76]: Very good point and should be there indeed! I tend to export this and then forget about it...

File: src/cloudsc_fortran/CMakeLists.txt [hunk 3]
```diff
@@ -15,25 +15,16 @@ ecbuild_add_option( FEATURE CLOUDSC_FORTRAN
...
             cloudsc_driver_mod.F90
             cloudsc.F90
+        LIBS
+            cloudsc-common-lib
         DEFINITIONS ${CLOUDSC_DEFINITIONS}
     )
-    target_link_libraries( dwarf-cloudsc-fortran PRIVATE cloudsc-common-lib )
-
-    target_include_directories(dwarf-cloudsc-fortran PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-    target_include_directories(dwarf-cloudsc-fortran PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/../common)
```
[the remark this answers is not in the document]
Comment (reuterbal) [reply src/cloudsc_fortran/CMakeLists.txt:28]: Well, the clean-up was in my yard, so ...

File: src/cloudsc_gpu/CMakeLists.txt [hunk 4]
```diff
@@ -45,6 +45,14 @@ if( HAVE_CLOUDSC_GPU_CLAW )
             cloudsc-common-lib
         DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_CLAW
     )
+
+    ecbuild_add_test(
+        TARGET dwarf-cloudsc-gpu-claw-serial
+        COMMAND bin/dwarf-cloudsc-gpu-claw
+        ARGS 1 1280 128
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
```
[the remark this answers is not in the document]
Comment (reuterbal) [reply src/cloudsc_gpu/CMakeLists.txt:53]: I'll see which way it works or if it can be resolved - otherwise I'll remove this test until fixed

**Comment by reuterbal** [comment]
Thanks for the helpful comments and tests! Indeed, the environment configuration was missing from the SCC tests.
I'm not sure what is wrong with the CLAW variant but wasn't able to fix it ad-hoc. This should be done in a separate PR I think and I have marked the test as ENABLED=OFF in the meantime.

ecbuild [looks for](https://github.com/ecmwf/ecbuild/blob/develop/cmake/ecbuild_add_test.cmake#L204) `mpiexec mpirun lamexec srun` (in that order) and which of those work seems to be dependent on the environment. On Atos, mpiexec is found and working fine, however, then serial runs of MPI-enabled builds fail without `srun -n1`, or similar, but work fine if built without MPI. With CMake variable `MPIEXEC_EXECUTABLE` the launcher can be overwritten but since this seems so highly environment-dependent, I'm not sure I'd attempt to get this working in every instance or even describe this for something like CLOUDSC.

**Comment by reuterbal** [comment]
PS: I had to piggy-back some fixes for the Github workflow, where CMake seems to have unearthed a GFortran 12 compiler in the runner image which isn't officially documented and doesn't seem to have a libm....

**Review by mlange05 — APPROVED** [review]
Excellent, many thanks! Works nicely now on multiple systems.

**Pull request merged**